src/test/objectstore/test_bluefs.cc

```diff
@@ -247,135 +246,81 @@ void join_all(std::vector<std::thread>& v)
 #define NUM_SINGLE_FILE_WRITERS 1
 #define NUM_MULTIPLE_FILE_WRITERS 2
 
-TEST(BlueFS, test_flush_1) {
-  uint64_t size = 1048576 * 128;
-  string fn = get_temp_bdev(size);
-  g_ceph_context->_conf.set_val(
-    "bluefs_alloc_size",
-    "65536");
-  g_ceph_context->_conf.apply_changes(nullptr);
-
-  BlueFS fs(g_ceph_context);
-  ASSERT_EQ(0, fs.add_block_device(BlueFS::BDEV_DB, fn, false));
-  fs.add_block_extent(BlueFS::BDEV_DB, 1048576, size - 1048576);
-  uuid_d fsid;
-  ASSERT_EQ(0, fs.mkfs(fsid));
-  ASSERT_EQ(0, fs.mount());
+TEST_F(BlueFSTest, test_flush_1) {
   {
     std::vector<std::thread> write_thread_multiple;
     uint64_t effective_size = size - (32 * 1048576); // leaving the last 32 MB for log compaction
     uint64_t per_thread_bytes = (effective_size/(NUM_MULTIPLE_FILE_WRITERS + NUM_SINGLE_FILE_WRITERS));
     for (int i=0; i<NUM_MULTIPLE_FILE_WRITERS ; i++) {
-      write_thread_multiple.push_back(std::thread(write_data, std::ref(fs), per_thread_bytes));
+      write_thread_multiple.push_back(std::thread(write_data, std::ref(*fs), per_thread_bytes));
     }
 
-    create_single_file(fs);
+    create_single_file(*fs);
     std::vector<std::thread> write_thread_single;
     for (int i=0; i<NUM_SINGLE_FILE_WRITERS; i++) {
-      write_thread_single.push_back(std::thread(write_single_file, std::ref(fs), per_thread_bytes));
+      write_thread_single.push_back(std::thread(write_single_file, std::ref(*fs), per_thread_bytes));
     }
 
     join_all(write_thread_single);
     join_all(write_thread_multiple);
   }
-  fs.umount();
-  rm_temp_bdev(fn);
 }
 
-TEST(BlueFS, test_flush_2) {
-  uint64_t size = 1048576 * 256;
```

Comment:
you're changing the size with your patch. I'm not sure whether it's critical but your new infrastructure provides no means to adjust such stuff on-demand. This is definitely some flexibility loss and a bit of cheating.